Ical.Net/Evaluation/RecurrenceUtil.cs

```diff
@@ -13,8 +14,12 @@ namespace Ical.Net.Evaluation;
 
 internal class RecurrenceUtil
 {
-    public static HashSet<Occurrence> GetOccurrences(IRecurrable recurrable, IDateTime dt, bool includeReferenceDateInResults) => GetOccurrences(recurrable,
-        new CalDateTime(dt.Date), new CalDateTime(dt.Date.AddDays(1).AddSeconds(-1)), includeReferenceDateInResults);
+    public static HashSet<Occurrence> GetOccurrences(IRecurrable recurrable, IDateTime dt, bool includeReferenceDateInResults)
+    {
+        var endTimeOnly = dt.Time?.Add(TimeSpan.FromSeconds(-1));
```

Comment:
InvalidOperationException

Reply:
Not sure, I understand. I think the problem still exists. If someone specifies a dt e.g. 20241203T000000, which is legit, then this code would lead to 20241203 T -000001.

Reply:
I think, we would introduce a new bug here and should consider fixing it right away.